code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -254,6 +254,110 @@
 /////////////////////////////////////////
 
 //Xeno acid smoke.
+
+/obj/effect/particle_effect/smoke/heal_smoke
+	name = "spore smoke"
+	time_to_live = 4
+	color = "#71ee96"
+	opacity = 0
+	anchored = 1
+	spread_speed = 7
+	smokeranking = SMOKE_RANK_HARMLESS
+
+	var/hivenumber = XENO_HIVE_NORMAL
+	var/gas_damage = 0
+
+/obj/effect/particle_effect/smoke/heal_smoke/Initialize(mapload, amount, datum/cause_data/cause_data)
+	var/mob/living/carbon/Xenomorph/X = cause_data.resolve_mob()
+	if (istype(X) && X.hivenumber)
+		hivenumber = X.hivenumber
+
+		set_hive_data(src, hivenumber)
+
+	. = ..()
+
+/obj/effect/particle_effect/smoke/heal_smoke/Crossed(mob/living/carbon/M as mob)
+	return
+
+/obj/effect/particle_effect/smoke/heal_smoke/affect(var/mob/living/carbon/M)
+	..()
+
+	if(isYautja(M))
+		return
+
+	if(M.stat == DEAD)
+		return
+
+	if(HAS_TRAIT(M, TRAIT_NESTED) && M.status_flags & XENO_HOST)
+		return
+
+	if(ishuman(M))
+		return
+
+	if(!M.ally_of_hivenumber(hivenumber))
+		return
+
+	if(isXeno(M))
+		var/mob/living/carbon/Xenomorph/X = M
+		to_chat(M, SPAN_DANGER("The spore cloud closes your wounds!"))
+		X.gain_health(100)
+
+
+/obj/effect/particle_effect/smoke/shield_smoke
+	name = "spore smoke"
+	time_to_live = 4
+	color = "#fcf700"
+	opacity = 1
+	anchored = 1
+	spread_speed = 7
+	smokeranking = SMOKE_RANK_HARMLESS
+
+	var/hivenumber = XENO_HIVE_NORMAL
+	var/gas_damage = 0
+
+/obj/effect/particle_effect/smoke/shield_smoke/Initialize(mapload, amount, datum/cause_data/cause_data)
+	var/mob/living/carbon/Xenomorph/X = cause_data.resolve_mob()
+	if (istype(X) && X.hivenumber)
+		hivenumber = X.hivenumber
+
+		set_hive_data(src, hivenumber)
+
+	. = ..()
+
+/obj/effect/particle_effect/smoke/shield_smoke/Crossed(mob/living/carbon/M as mob)
+	return
+
+/obj/effect/particle_effect/smoke/shield_smoke/affect(var/mob/living/carbon/M)
+	..()
+
+	if(isYautja(M))
+		return
+
+	if(M.stat == DEAD)
+		return
+
+	if(HAS_TRAIT(M, TRAIT_NESTED) && M.status_flags & XENO_HOST)
+		return
+
+	if(ishuman(M))
+		return
+
+	if(!M.ally_of_hivenumber(hivenumber))
+		return
```

Comment:
see above comment

also would be wise to make this into a proc on the parent to avoid duplicated code